### PR TITLE
jira -> users table initial

### PR DESCRIPTION
**Pull request by lwsanty**
relates to [[DEV-1039]]
relates to [[DEV-1040]]

Signed-off-by: lwsanty <lwsanty@gmail.com>

[DEV-1039]: https://athenianco.atlassian.net/browse/DEV-1039
[DEV-1040]: https://athenianco.atlassian.net/browse/DEV-1040

### CODE REVIEW

File: sql/migrations/035_jira_user.up.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+BEGIN;
+
+ALTER TABLE jira.issue ADD COLUMN creator_id text;
+UPDATE jira.issue SET creator_id = 'needs re-fetch';
+ALTER TABLE jira.issue ALTER COLUMN creator_id SET NOT NULL;
```
Comment (vmarkovtsev) [inline sql/migrations/035_jira_user.up.sql:5]: Shall we consider renaming "creator" to "reporter", as it is in the JIRA UI? Or does it come directly from JIRA API?

File: sql/migrations/035_jira_user.up.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+UPDATE jira.issue SET creator_display_name = 'needs re-fetch';
+ALTER TABLE jira.issue ALTER COLUMN creator_display_name SET NOT NULL;
+
+ALTER TABLE jira.issue ADD COLUMN assignee_id text;
+ALTER TABLE jira.issue ADD COLUMN assignee_display_name text;
+
+CREATE TABLE jira.user (
+    acc_id bigint NOT NULL,
+    id text NOT NULL,
+    type text NOT NULL,
```
Comment (vmarkovtsev) [inline sql/migrations/035_jira_user.up.sql:17]: What's the user type?

**Review by vmarkovtsev — APPROVED** [review]

File: sql/migrations/035_jira_user.up.sql (hunk 1 above)
Comment (lwsanty) [reply sql/migrations/035_jira_user.up.sql:5]: ouch I have missed that these are different things https://community.atlassian.com/t5/Jira-questions/creator-vs-reporter/qaq-p/1028179

File: sql/migrations/035_jira_user.up.sql (hunk 2 above)
Comment (lwsanty) [reply sql/migrations/035_jira_user.up.sql:17]: either atlassian or app

File: sql/migrations/035_jira_user.up.sql (hunk 1 above)
Comment (lwsanty) [reply sql/migrations/035_jira_user.up.sql:5]: I will set reporter then

**Comment by vmarkovtsev** [comment]
@lwsanty The priority display names in the issue table should be saved lower case (just double-checking).

**Comment by vmarkovtsev** [comment]
Update: no need for lower case, I glitched.

File: sql/migrations/035_jira_user.up.sql [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+-- TODO: commenters IDs
+ALTER TABLE jira.issue ADD COLUMN commenters_display_names text[];
+
+-- TODO: priority ID
+ALTER TABLE jira.issue ADD COLUMN priority_name text;
+UPDATE jira.issue SET priority_name = 'needs re-fetch';
+ALTER TABLE jira.issue ALTER COLUMN priority_name SET NOT NULL;
+
+CREATE INDEX IF NOT EXISTS issue_reporter ON jira.issue(acc_id, created, resolved, lower(priority_name), lower(reporter_display_name));
+CREATE INDEX IF NOT EXISTS issue_resolver ON jira.issue(acc_id, created, resolved, lower(priority_name), lower(assignee_display_name));
```
Comment (vmarkovtsev) [inline sql/migrations/035_jira_user.up.sql:23]: Having spent a few hours coding sample SQL, these seem to be needed:
```suggestion
CREATE INDEX IF NOT EXISTS issue_reporter ON jira.issue(acc_id, created, resolved, lower(reporter_display_name), lower(priority_name));
CREATE INDEX IF NOT EXISTS issue_resolver ON jira.issue(acc_id, created, resolved, lower(assignee_display_name), lower(priority_name));
CREATE INDEX IF NOT EXISTS issue_commenters ON jira.issue USING GIN(acc_id, created, resolved, commenters_display_names, lower(priority_name));
CREATE INDEX IF NOT EXISTS issue_reporter_active ON jira.issue(acc_id, updated, created, resolved,  lower(reporter_display_name), lower(priority_name));
CREATE INDEX IF NOT EXISTS issue_resolver_active ON jira.issue(acc_id, updated, created, resolved, lower(assignee_display_name), lower(priority_name));
CREATE INDEX IF NOT EXISTS issue_commenters_updated ON jira.issue USING GIN(acc_id, updated, created, resolved, commenters_display_names, lower(priority_name));
```
IDK much about GIN, we really need Denys to adjust here... If it does not work with GIN then we can add it later.

File: sql/migrations/035_jira_user.up.sql [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+ALTER TABLE jira.issue ADD COLUMN reporter_display_name text;
+UPDATE jira.issue SET reporter_display_name = 'needs re-fetch';
+ALTER TABLE jira.issue ALTER COLUMN reporter_display_name SET NOT NULL;
+
+ALTER TABLE jira.issue ADD COLUMN assignee_id text;
+ALTER TABLE jira.issue ADD COLUMN assignee_display_name text;
+
+-- TODO: commenters IDs
+ALTER TABLE jira.issue ADD COLUMN commenters_display_names text[];
```
Comment (vmarkovtsev) [inline sql/migrations/035_jira_user.up.sql:15]: These should be written lower case, ok?

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: sql/migrations/035_jira_user.up.sql (hunk 3 above)
Comment (lwsanty) [reply sql/migrations/035_jira_user.up.sql:23]: agreed to skip for now

File: sql/migrations/035_jira_user.up.sql (hunk 4 above)
Comment (lwsanty) [reply sql/migrations/035_jira_user.up.sql:15]: Done.

File: sql/migrations/035_jira_user.up.sql [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ALTER TABLE jira.issue ADD COLUMN priority_id text;
+UPDATE jira.issue SET priority_id = 'needs re-fetch';
+ALTER TABLE jira.issue ALTER COLUMN priority_id SET NOT NULL;
+
+ALTER TABLE jira.issue ADD COLUMN priority_name text;
+UPDATE jira.issue SET priority_name = 'needs re-fetch';
+ALTER TABLE jira.issue ALTER COLUMN priority_name SET NOT NULL;
+
+CREATE INDEX IF NOT EXISTS issue_reporter ON jira.issue(acc_id, created, resolved, lower(priority_name), lower(reporter_display_name));
+CREATE INDEX IF NOT EXISTS issue_resolver ON jira.issue(acc_id, created, resolved, lower(priority_name), lower(assignee_display_name));
```
Comment (vmarkovtsev) [inline sql/migrations/035_jira_user.up.sql:26]: The indexes are still the old ones here. I've proposed 6, 4 are normal ones and 2 are GIN-s that we drop.

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: sql/migrations/035_jira_user.up.sql (hunk 5 above)
Comment (lwsanty) [reply sql/migrations/035_jira_user.up.sql]: Done.

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**